framework/web/migrations/m160313_153426_session_init.php

```diff
@@ -20,16 +20,23 @@ class m160313_153426_session_init extends Migration
      */
     public function up()
     {
+        $dataType = $this->binary();
         $tableOptions = null;
-        if ($this->db->driverName === 'mysql') {
-            // http://stackoverflow.com/questions/766809/whats-the-difference-between-utf8-general-ci-and-utf8-unicode-ci
-            $tableOptions = 'CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE=InnoDB';
+
+        switch ($this->db->driverName) {
+            case 'mysql':
+                // http://stackoverflow.com/questions/766809/whats-the-difference-between-utf8-general-ci-and-utf8-unicode-ci
+                $tableOptions = 'CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE=InnoDB';
+                break;
+            case 'sqlsrv':
+                $dataType = $this->text();
```

Comment:
we had issues with non-utf8 data stored in session, which caused text to fail so we used binary for that. Is this covered by a test?

Reply:
I remembered it, but was not sure about characters that cause that behavior, so typed something random: https://github.com/yiisoft/yii2/pull/15049/files#diff-722a201958cf84ca29390026ac2a10eaR138

Hope it covers. If you remember something more exact – share a link please

Reply:
The issue is about raw binary data that does not follow utf8 encoding, e.g. https://github.com/yiisoft/yii2/issues/933